library/src/blas_ex/rocblas_gemm_ex.hpp

```diff
@@ -660,8 +660,8 @@ rocblas_status gemm_ex_batched_template(rocblas_handle    handle,
     rocblas_stride    stride_i;
 
     if(!arch_lt906 && (std::is_same<Ti, float>{} || std::is_same<Ti, double>{})
-       && ((ldc >= ldd && stride_c >= stride_d && m == ldd)
-           || (ldc == ldd && stride_c == stride_d)))
+       && ((ldc >= ldd && stride_c >= stride_d && m == ldd) || (ldc == ldd && stride_c == stride_d)
+           || batch_count == 1))
```

Comment:
For this test the ldc vs ldd check must still be honored in the batch_count == 1 case I would think.    So don't you need the batch count == 1 case within each (stride test || batch_count == 1) or fix the stride setting to be the same as before?